File: AD-HOC/1800/main.rb
require "set"

q = gets.split.first.to_i
s = gets.split.map(&:to_i).to_set
q.times do
  office = gets.to_i
  if s.include?(office)
    puts 0
  else
    puts 1
    s.add(office)
  end
end

File: AD-HOC/1800/main.dart
import 'dart:io';

void main() {
  int q = int.parse(stdin.readLineSync()!.split(' ').first);
  Set<int> s = stdin.readLineSync()!.split(' ').map(int.parse).toSet();
  while (q-- > 0) {
    int office = int.parse(stdin.readLineSync()!);
    if (s.contains(office)) {
      print(0);
    } else {
      print(1);
      s.add(office);
    }
  }
}
